libs/platform/src/lib/components/form/checkbox/checkbox.component.ts

```diff
@@ -115,6 +139,7 @@ export class CheckboxComponent extends BaseInput implements AfterViewInit {
         private _ngZone: NgZone
     ) {
         super(_changeDetector, ngControl, ngForm);
+        this.name = `fdp-checkbox-${nextUniqueId++}`;
```

Comment:
Why are we setting `name` here? Isn't it an `@Input` field inherited from BaseComponent?

Reply:
in declarative checkbox-group, fdp-checkbox is passed under fdp-checkbox-group. so fdp-checkbox  is trying to create component, but id and name will not be available, and it fails to create because of baseInput check.
id and name are passed for fdp-checkbox-group not for fdp-checkbox.
So have to give some dummy value for id and name for fdp-checkbox. so component creation happens.
upon creation of fdp-checkbox-group, these get over written.

Reply:
Hmm ... I feel like this would be a common problem with all of our components. I guess this is OK for now, but we should take a look at solving this problem at the BaseComponent or BaseInput level.